packages/@aws-cdk/aws-ecs/README.md

```diff
@@ -1,2 +1,240 @@
+
+```ts
+// Create an ECS cluster
+const cluster = new ecs.EcsCluster(this, 'Cluster', {
+  vpc,
+});
+
+// Add capacity to it
+cluster.addDefaultAutoScalingGroupCapacity({
+  instanceType: new InstanceType("t2.xlarge"),
+  instanceCount: 3,
+});
+
+// Instantiate ECS Service with an automatic load balancer
+const ecsService = new ecs.LoadBalancedEc2Service(this, 'Service', {
+  cluster,
+  memoryLimitMiB: 512,
+  image: ecs.DockerHub.image("amazon/amazon-ecs-sample"),
+});
+```
+
+### Fargate vs ECS
+
+There are two sets of constructs in this library; one to run tasks on ECS and
+one to run Tasks on Fargate.
+
+- Use the `Ec2TaskDefinition` and `Ec2Service` constructs to run tasks on EC2 instances running in your account.
+- Use the `FargateTaskDefinition` and `FargateService` constructs to run tasks on
+  instances that are managed for you by AWS.
+
+Here are the main differences:
+
+- **EC2**: instances are under your control. Complete control of task to host
+  allocation. Required to specify at least a memory reseration or limit for
+  every container. Can use Host, Bridge and AwsVpc networking modes. Can attach
+  Classic Load Balancer. Can share volumes between container and host.
+- **Fargate**: tasks run on AWS-managed instances, AWS manages task to host
+  allocation for you. Requires specification of memory and cpu sizes at the
+  taskdefinition level. Only supports AwsVpc networking modes and
+  Application/Network Load Balancers. Only the AWS log driver is supported.
+  Many host features are not supported such as adding kernel capabilities
+  and mounting host devices/volumes inside the container.
+
+For more information on EC2 vs Fargate and networking see the AWS Documentation:
+[AWS Fargate](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/AWS_Fargate.html) and
+[Task Networking](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-networking.html).
+
+### Clusters
+
+An `EcsCluster` defines the infrastructure to run your
+tasks on. You can run many tasks on a single cluster.
+
+To create a cluster that can run Fargate tasks, go:
+
+```ts
+const cluster = new ecs.EcsCluster(this, 'Cluster', {
+  vpc: vpc
+});
+```
+
+If you wish to use tasks with EC2 launch-type, you also have to add capacity to
+your cluster in order for tasks to be scheduled on your instances.  Typically,
+you will add an AutoScalingGroup with instances running the latest
+ECS-optimized AMI to the cluster. There is a method to build and add such an
+AutoScalingGroup automatically, or you can supply a customized AutoScalingGroup
+that you construct yourself. It's possible to add multiple AutoScalingGroups
+with various instance types if you want to.
+
+Creating an ECS cluster and adding capacity to it looks like this:
+
+```ts
+const cluster = new ecs.EcsCluster(this, 'Cluster', {
+  vpc: vpc
+});
+
+// Either add default capacity
+cluster.addDefaultAutoScalingGroupCapacity({
```

Comment:
Maybe instead of "default" we should call this "standard" or even "EcsOptimized"